code/game/objects/items/explosives/mine.dm

```diff
@@ -78,13 +78,23 @@ Stepping directly on the mine will also blow it up
 	user.visible_message(span_notice("[user] finishes deploying [src]."), \
 	span_notice("You finish deploying [src]."))
 	var/obj/item/card/id/id = user.get_idcard()
-	iff_signal = id?.iff_signal
+	var/iff_sig = id?.iff_signal
+	deploy_mine(user, iff_sig)
+
+///this proc is used to deploy a mine
+/obj/item/explosive/mine/proc/deploy_mine(mob/living/user, iff_sig)
+	var/mine_dir
```

Comment:
this is a complete nitpick, but in programming in general you want to declare your variable as late as possible. 

This is to improve readability, so here you'd want to declare it line 92